src/service/resources-impl.js

```diff
@@ -229,6 +230,17 @@ export class ResourcesImpl {
      */
     this.intersectionObserverCallbackFired_ = false;
 
+    /** @private {WeakMap<!Element, boolean>} */
+    this.inViewportMap_ = new WeakMap();
```

Comment:
Ideally we don't have any of this either. Do we really need this code?

Reply:
I'm hoping we can get rid of it. See https://github.com/ampproject/amphtml/pull/30973

Reply:
Gone!

Reply:
deferring!